cms/djangoapps/contentstore/tests/test_libraries.py

```diff
@@ -348,31 +349,58 @@ def test_change_after_first_sync(self):
 
         # Add a LibraryContent block to the course:
         lc_block = self._add_library_content_block(course, self.lib_key)
-        lc_block = self._refresh_children(lc_block)
+        lc_block = self._upgrade_and_sync(lc_block)
+        good_library_id = lc_block.source_library_id
+        good_library_version = lc_block.source_library_version
+        assert good_library_id
+        assert good_library_version
         self.assertEqual(len(lc_block.children), 1)
+        self.assertEqual(modulestore().get_item(lc_block.children[0]).data, data_value)
 
         # Now, change the block settings to have an invalid library key:
+        bad_library_id = "library-v1:NOT+FOUND"
         resp = self._update_block(
             lc_block.location,
-            {"source_library_id": "library-v1:NOT+FOUND"},
+            {"source_library_id": bad_library_id},
         )
         self.assertEqual(resp.status_code, 200)
+
         lc_block = modulestore().get_item(lc_block.location)
-        self.assertEqual(len(lc_block.children), 1)  # Children should not be deleted due to a bad setting.
-        html_block = modulestore().get_item(lc_block.children[0])
-        self.assertEqual(html_block.data, data_value)
+        # Source library id should be set to the new bad one...
+        assert lc_block.source_library_id == bad_library_id
+        # ...but old source library version should be preserved...
+        assert lc_block.source_library_version == good_library_version
+        # ...and children should not be deleted due to a bad setting.
+        self.assertEqual(len(lc_block.children), 1)
+        self.assertEqual(modulestore().get_item(lc_block.children[0]).data, data_value)
 
-    def test_refreshes_children_if_libraries_change(self):
-        """ Tests that children are automatically refreshed if libraries list changes """
+        # Attempting to force an upgrade (the user would have to do this through the API, as
+        # the UI wouldn't give them the option) returns a 400 and preserves the LC block's state.
+        self._upgrade_and_sync(lc_block, status_code_expected=400)
+
+        # (Repeat the exact same checks)
+        lc_block = modulestore().get_item(lc_block.location)
+        # Source library id should be set to the new bad one...
+        assert lc_block.source_library_id == bad_library_id
+        # ...but old source library version should be preserved...
+        assert lc_block.source_library_version == good_library_version
+        # ...and children should not be deleted due to a bad setting.
+        self.assertEqual(len(lc_block.children), 1)
+        self.assertEqual(modulestore().get_item(lc_block.children[0]).data, data_value)
+
+    def test_sync_if_source_library_changed(self):
+        """
+        Tests that children are automatically synced with new lib if source library id is changed.
+        """
```

Comment:
This block was originally:
```python
    def test_refreshes_children_if_libraries_change(self):
        """ Tests that children are automatically refreshed if libraries list changes """
```
which was just not an accurate description of what it tests.